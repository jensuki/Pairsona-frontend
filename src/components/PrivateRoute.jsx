import { Navigate } from 'react-router-dom';
import { useContext } from 'react';
import UserContext from '../context/UserContext';

// eslint-disable-next-line react/prop-types
const PrivateRoute = ({ children }) => {
    const { currUser } = useContext(UserContext);

    return currUser ? children : <Navigate to="/login" replace />;
};

export default PrivateRoute;
