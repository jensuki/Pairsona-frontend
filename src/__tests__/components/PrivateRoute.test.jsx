import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { UserProvider } from '../../test_common/setupTests';
import { demoUser } from '../../test_common/mockData';
import PrivateRoute from '../../components/PrivateRoute';
import { expect, test, describe } from 'vitest';

describe('PrivateRoute component', () => {
    test('renders without crashing', () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <PrivateRoute />
                </UserProvider>
            </MemoryRouter>
        )
    })

    test('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <PrivateRoute>
                        <div>Private Content</div>
                    </PrivateRoute>
                </UserProvider>
            </MemoryRouter>
        )
        expect(asFragment()).toMatchSnapshot();
    })

    test('renders children when user is authenticated', () => {
        render(
            <MemoryRouter>
                <UserProvider currUser={demoUser}>
                    <PrivateRoute>
                        <div>Private Content</div>
                    </PrivateRoute>
                </UserProvider>
            </MemoryRouter>
        )
        // verify private content is rendered while demoUser is authenticated
        expect(screen.getByText('Private Content')).toBeInTheDocument();
    })

    test('redirects to login when user not authenticated', () => {
        render(
            <MemoryRouter initialEntries={['/private']}>
                <Routes>
                    <Route path="/login" element={<div>Log In</div>} />
                    <Route
                        path="/private"
                        element={
                            <UserProvider currUser={null}>
                                <PrivateRoute>
                                    <div>Private Content</div>
                                </PrivateRoute>
                            </UserProvider>
                        }
                    />
                </Routes>
            </MemoryRouter>
        );
        // ensure private content isnt rendered
        expect(screen.queryByText('Private Content')).not.toBeInTheDocument();

        // verify redirection to /login
        expect(screen.getByText('Log In')).toBeInTheDocument();
    });

})