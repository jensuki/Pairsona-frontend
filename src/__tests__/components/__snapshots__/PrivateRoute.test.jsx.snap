// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrivateRoute component > matches snapshot 1`] = `<DocumentFragment />`;
